### DISCUSSION TITLE
Will there ever be support for fixed ring descriptors?

### DISCUSSION THREAD

**Discussion by arajni3** [axboe/liburing#1024] [Q&A] [answered]
We currently have registered ring descriptors but those are meant only for io_uring_enter(), thus the family of io_uring_prep_msg_ring* functions is limited to using regular ring descriptors.

**Comment by axboe** [comment, marked as answer]
You cannot register an io_uring fd, the ring registration itself is special and does not use or participate in the normal fixed file registration. This means that you could not have the destination ring be a fixed file, ever, it's simply not supported. For the source ring, that is simply managed by having that ring fd registered with ```io_uring_register_ring_fd()```, in which case no fget/fput will happen for that side.
